servicetest/timingprofiling/test_profiling.py

```diff
@@ -17,307 +17,181 @@
-                softwarecontainer-agent
-            """
-            print "SoftwareContainer-agent has died for some reason"
-            rec.terminate()
-            sys.exit(-1)
+# This function is used by the 'agent' fixture to know where the log should be stored
+def logfile_path():
+    return CURRENT_DIR + "/test.log"
+
 
-        apps = []
-        for app in range(0, numStarts):
+def run_test(num_starts=3):
+        apps = list()
+        for app in range(0, num_starts):
             """ Start numStarts apps in softwarecontainer.
             """
             print "Start app " + str(app)
-            container = ContainerApp()
-            container.start()
+            container = Container()
+
+            # A basic container configuration, content is not important for this test.
+            container_data = {
+                Container.PREFIX: "profiling-test-",
+                Container.CONFIG: "{enableWriteBuffer: false}",
+                Container.BIND_MOUNT_DIR: "app",
+                Container.HOST_PATH: CURRENT_DIR
+            }
+            container.start(container_data)
+
+            # A minimal gateway config so the gateway can be configured and enabled.
+            dbus_gw_config = [{
+                "dbus-gateway-config-session": [{
+                    "direction": "*",
+                    "interface": "*",
+                    "object-path": "*",
+                    "method": "*"
+                }],
+                "dbus-gateway-config-system": []
+            }]
+            container.set_gateway_config("dbus", dbus_gw_config)
+
+            container.launch_command("/gateways/app/simple")
             apps.append(container)
 
         time.sleep(5)
 
+        # Tear down all started apps
         for app in apps:
-            """ Tear down all started apps
-            """
             app.terminate()
 
-        agent.terminate()
-        rec.terminate()
-
 
-def removeAnsi(astring):
+def remove_ansi(astring):
     ansi_escape = re.compile(r'\x1b[^m]*m')
     return ansi_escape.sub('', astring)
 
 
-def getPythonPoint(logFile, pointName, matchNumber=1):
-    """
-    This function will grab the timestamp from a python log entry with pointName in the
-    logFile
+def get_python_point(log_file, point_name, match_number=1):
+    """ This function will grab the timestamp from a python log entry with pointName in
+        the log_file
     """
     match = 0
-    logFile.seek(0)
-    for line in logFile:
-        if re.search(pointName, line):
+    log_file.seek(0)
+    for line in log_file:
+        if re.search(point_name, line):
             match = match + 1
-            if match == matchNumber:
-                return removeAnsi(string.split(line)[2])
-    print "Nothing found! " + pointName + " " + str(matchNumber)
+            if match == match_number:
+                return remove_ansi(string.split(line)[2])
+    print "Nothing found! " + point_name + " " + str(match_number)
```

Comment:
We've never used any of our Python code with Python3 ever. Usually, that's not what's available and preferred in various contexts. Maybe that's changing, I'm not sure. I'm not sure I'd like to start checking our stuff for compatibility in general, e.g. use of bindings and whatnot.